### PR TITLE
user route test, subscriber service, schema test

### PR DESCRIPTION
**Pull request by WeiyuSun**
1. user(include subscriber) route and controller test
2. subscriber service and schema test
3. one bug find: for the endpoint /user/subscription/setNotification, The request type is POST. But I use Request Parameter to handle the parameters. I have changed to request body. see the code details
4. One question is: some code(like line98 below) in the controller is just used to prevent unknow errors. I think the code cannot be reached if the database and server run normally.  So the test coverage cannot reach 100%. Is this OK? or do I have to delete some code for 100% coverage? 
image
5. @DaTanUmanitoba Can you help me to test this function? 
image

### CODE REVIEW

**Comment by DaTanUmanitoba** [comment]
Sure, I will write the tests for notification for subscribers.

**Review by DaTanUmanitoba — DISMISSED** [review]
It seems all good.

File: backend/middleware/user-middleware.js [hunk 1]
```diff
@@ -153,9 +153,9 @@ function unsubscriptionValidation(req, res, next) {
 }
 
 function setNotificationValidation(req, res, next) {
-    let creatorId = req.query.creator_id
-    let audienceId = req.query.user_id
-    let setting = req.query.set_notification;
+    let creatorId = req.body.creator_id
```
Comment (ShahzaibParacha) [inline backend/middleware/user-middleware.js:156]: all else looks good, I think @orangeboy55555 will need to approve if this needs to be body or query

**Review by ShahzaibParacha — CHANGES_REQUESTED** [review]

File: backend/middleware/user-middleware.js (hunk 1 above)
Comment (orangeboy55555) [reply backend/middleware/user-middleware.js:156]: I think it should be query since GET requests don't have a body.

**Review by orangeboy55555 — CHANGES_REQUESTED** [review]
I think you should delete the unreachable code. They want 100% code coverage XD

**Review by DaTanUmanitoba — APPROVED** [review]
seems all good to merge

**Review by orangeboy55555 — APPROVED** [review]
Looks good

**Review by ShahzaibParacha — APPROVED** [review]

**Pull request merged**